scripts/Muon/GUI/Common/load_run_widget/presenter.py

```diff
@@ -0,0 +1,320 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import (absolute_import, division, print_function)
+
+import copy
+
+from Muon.GUI.Common import thread_model
+import Muon.GUI.Common.utilities.run_string_utils as run_utils
+import Muon.GUI.Common.utilities.muon_file_utils as file_utils
+import Muon.GUI.Common.utilities.algorithm_utils as algorithm_utils
+from Muon.GUI.Common.ADSHandler.muon_workspace_wrapper import MuonWorkspaceWrapper
+
+
+class LoadRunWidgetPresenter(object):
+    def __init__(self, view, model):
+        self._view = view
+        self._model = model
+        self._load_thread = None
+
+        self._load_multiple_runs = False
+        self._use_threading = True
+        self._multiple_file_mode = "Co-add"
+
+        self._instrument = ""
+        self._view.set_current_instrument(self._instrument)
+
+        self._set_connections()
+
+    def _set_connections(self):
+        self._view.on_load_current_run_clicked(self.handle_load_current_run)
+        self._view.on_increment_run_clicked(self.handle_increment_run)
+        self._view.on_decrement_run_clicked(self.handle_decrement_run)
+        self._view.on_run_edit_changed(self.handle_run_changed_by_user)
+
+    def show(self):
+        self._view.show()
+
+    # used with threading
+    def create_load_thread(self):
```

Comment:
this seems to be always assigned to a member of the class. We dont we do that here and then make it a void?

Reply:
I think the code is clearer in it's current form and we don't really gain anything by assigning within this function?

Reply:
I cant type today. I just wanted to know why it was done this way